docs/api-reference/next/link.md

```diff
@@ -42,37 +53,38 @@ export default Home
 
 `Link` accepts the following props:
 
-- `href` - The path inside `pages` directory. This is the only required prop
-- `as` - The path that will be rendered in the browser URL bar. Used for dynamic routes
+- `href` - The path or URL to navigate to. This is the only required prop
+- `as` - Optional decorator for the path that will be shown in the browser URL bar. Used for dynamic routes before Next.js 9.5.3
 - [`passHref`](#if-the-child-is-a-custom-component-that-wraps-an-a-tag) - Forces `Link` to send the `href` property to its child. Defaults to `false`
-- `prefetch` - Prefetch the page in the background. Defaults to `true`. Any `<Link />` that is in the viewport (initially or through scroll) will be preloaded. Pages using [Static Generation](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) will preload `JSON` files with the data for faster page transitions.
+- `prefetch` - Prefetch the page in the background. Defaults to `true`. Any `<Link />` that is in the viewport (initially or through scroll) will be preloaded. Pages using [Static Generation](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) will preload `JSON` files with the data for faster page transitions
 - [`replace`](#replace-the-url-instead-of-push) - Replace the current `history` state instead of adding a new url into the stack. Defaults to `false`
 - [`scroll`](#disable-scrolling-to-the-top-of-the-page) - Scroll to the top of the page after a navigation. Defaults to `true`
 - [`shallow`](/docs/routing/shallow-routing.md) - Update the path of the current page without rerunning [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation), [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering) or [`getInitialProps`](/docs/api-reference/data-fetching/getInitialProps.md). Defaults to `false`
 
-External URLs, and any links that don't require a route navigation using `/pages`, don't need to be handled with `Link`; use the anchor tag for such cases instead.
+## If the route has dynamic segments
 
-## Dynamic routes
+There is nothing special to do when linking to a [dynamic route](/docs/routing/dynamic-routes.md) (including [catch all routes](/docs/routing/dynamic-routes.md#catch-all-routes)). However, it can become quite common and handy to use [interpolation](/docs/routing/introduction.md#linking-to-dynamic-paths) or an [URL Object](#with-url-object) to generate the link.
```

Comment:
```suggestion
There is nothing special to do when linking to a [dynamic route](/docs/routing/dynamic-routes.md) (including [catch all routes](/docs/routing/dynamic-routes.md#catch-all-routes)) since Next.js 9.5.3. However, it can become quite common and handy to use [interpolation](/docs/routing/introduction.md#linking-to-dynamic-paths) or an [URL Object](#with-url-object) to generate the link.
```

Should we still link to a previous version of the docs that show what was required for `href` and `as` for dynamic routes before v9.5.3 here?

Reply:
Yeah that should probably be done given that otherwise users will get unexpected behavior.

Reply:
Done!